samples/net/wifi/shell/boards/mimxrt1170_evk_mimxrt1176_cm7_A.conf

```diff
@@ -0,0 +1,143 @@
+# Early Console and Debug Configuration
+CONFIG_EARLY_CONSOLE=y
+CONFIG_LOG=y
+CONFIG_LOG_MODE_IMMEDIATE=y
+# CONFIG_WIFI_LOG_LEVEL_DBG=y
+# CONFIG_NET_LOG=y
+# CONFIG_NET_SOCKETS_LOG_LEVEL_DBG=y
+# CONFIG_NET_MGMT_EVENT_LOG_LEVEL_DBG=y
+CONFIG_INIT_STACKS=y
+CONFIG_TEST_RANDOM_GENERATOR=y
+
+# OS and Memory Configuration
+CONFIG_THREAD_CUSTOM_DATA=y
+CONFIG_EVENTS=y
+CONFIG_SYS_HEAP_AUTO=y
+CONFIG_HEAP_MEM_POOL_SIZE=307200
+CONFIG_SCHED_MULTIQ=y
+CONFIG_ZVFS_OPEN_MAX=30
+CONFIG_POSIX_API=y
+CONFIG_BUILD_OUTPUT_HEX=y
+
+# Stack Size Configuration
+CONFIG_MAIN_STACK_SIZE=5200
+CONFIG_SHELL_STACK_SIZE=8192
+CONFIG_NET_TX_STACK_SIZE=4096
+CONFIG_NET_RX_STACK_SIZE=4096
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=4096
+CONFIG_NET_MGMT_EVENT_STACK_SIZE=4096
+CONFIG_NET_SOCKETS_SERVICE_STACK_SIZE=4096
+
+# WiFi Core Configuration
+CONFIG_WIFI=y
+CONFIG_WIFI_LOG_LEVEL_ERR=y
+CONFIG_WIFI_NXP=y
+CONFIG_NXP_88W8987=y
+CONFIG_NXP_88W8987_MURATA_1ZM_M2=y
```

Comment:
I don't *love* the current approach we use for module selection here (IE choosing it with a Kconfig). This is a property of the hardware- IMO ideally we should be using a different devicetree compatible for each module.

The other issue here is that this M.2 slot can support a variety of modules, not just the 1ZM. There is also an IW416 based module for example (1XK).